test/e2e/e2e_suite_test.go

```diff
@@ -146,29 +154,53 @@ var _ = SynchronizedBeforeSuite(func() []byte {
 
 	By("Initializing the bootstrap cluster")
 	vsphereframework.InitBootstrapCluster(ctx, bootstrapClusterProxy, e2eConfig, clusterctlConfigPath, artifactFolder)
-	namespaces = map[*corev1.Namespace]context.CancelFunc{}
+
+	ipamLabels := ipam.GetIPAddressClaimLabels()
+	var ipamLabelsRaw []string
+	for k, v := range ipamLabels {
+		ipamLabelsRaw = append(ipamLabelsRaw, fmt.Sprintf("%s=%s", k, v))
+	}
+
 	return []byte(
 		strings.Join([]string{
 			artifactFolder,
 			configPath,
 			clusterctlConfigPath,
 			bootstrapClusterProxy.GetKubeconfigPath(),
+			strings.Join(ipamLabelsRaw, ";"),
 		}, ","),
 	)
 }, func(data []byte) {
 	// Before each ParallelNode.
 	parts := strings.Split(string(data), ",")
-	Expect(parts).To(HaveLen(4))
+	Expect(parts).To(HaveLen(5))
 
 	artifactFolder = parts[0]
 	configPath = parts[1]
 	clusterctlConfigPath = parts[2]
 	kubeconfigPath := parts[3]
+	ipamLabelsRaw := parts[4]
+
+	namespaces = map[*corev1.Namespace]context.CancelFunc{}
+
+	By("Initializing the vSphere session to ensure credentials are working", initVSphereSession)
 
 	var err error
 	e2eConfig, err = vsphereframework.LoadE2EConfig(ctx, configPath)
 	Expect(err).NotTo(HaveOccurred())
 	bootstrapClusterProxy = framework.NewClusterProxy("bootstrap", kubeconfigPath, initScheme(), framework.WithMachineLogCollector(LogCollector{}))
+
+	ipamLabels := map[string]string{}
+	for _, s := range strings.Split(ipamLabelsRaw, ";") {
+		splittedLabel := strings.Split(s, "=")
+		Expect(splittedLabel).To(HaveLen(2))
+
+		ipamLabels[splittedLabel[0]] = splittedLabel[1]
+	}
+	if e2eIPAMKubeconfig != "" {
```

Comment:
If I see correctly this means the tests panic now in beforeeach if e2eIPAMKubeconfig is ""

Reply:
Based on the Implementation of ipam.New it looks like it's good enough to drop l.200